test/e2e/pkg/ssh/ssh.go

```diff
@@ -0,0 +1,82 @@
+package ssh
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+	"golang.org/x/crypto/ssh/knownhosts"
+)
+
+func Exec(cfg *ssh.ClientConfig, addr, cmd string) error {
+	c, err := ssh.Dial("tcp", addr, cfg)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+	s, err := c.NewSession()
+	if err != nil {
+		return err
+	}
+	defer s.Close()
+	err = s.Run(cmd)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func NewClientConfig() (*ssh.ClientConfig, error) {
+	ss := os.Getenv("SSH_AUTH_SOCK")
+	if ss == "" {
+		return nil, errors.New("SSH_AUTH_SOCK environment variable is empty. Is the ssh-agent running?")
+	}
+	c, err := net.Dial("unix", ss)
+	if err != nil {
+		return nil, err
+	}
+	ac := agent.NewClient(c)
+	hkc, err := knownhosts.New(filepath.Join(os.Getenv("HOME"), ".ssh", "known_hosts"))
+	if err != nil {
+		return nil, err
+	}
+	signers, err := ac.Signers()
+	if err != nil {
+		return nil, err
+	}
+	am := make([]ssh.AuthMethod, 0, len(signers))
+	for _, signer := range signers {
+		am = append(am, ssh.PublicKeys(signer))
+	}
+	return &ssh.ClientConfig{
+		User:              "root",
+		HostKeyCallback:   hkcWrapper(hkc),
+		Auth:              am,
+		HostKeyAlgorithms: []string{ssh.KeyAlgoED25519},
```

Comment:
Does this mean that keys always have to be ed25519?

Reply:
Yes, I intended to remove this but appear to have missed it. Removed. When not set it uses a sensible default.